### DISCUSSION TITLE
Released v0.19.0

### DISCUSSION THREAD

**Discussion by boojack** [usememos/memos#2844] [Announcements]
## Upgrade overview

⚠️ This release contains upgrade notes that deviate from the norm:

* Requires database migrations;
* Use short UUIDs instead of auto-incrementing IDs for memo and resource in API and frontend;
* The docker image tag `latest` will be deprecated, use `stable` instead;

## New Loaded

* Added new syntax for embedding content: `![[memos/memoid]]`;
* Supported rendering HTML directly with codeblock `__html`;
* Supported renaming tag;
* Supported collapse and expand navigator;

## What's Changed
* feat: add a webhook action for deleted memos by @ertuil in https://github.com/usememos/memos/pull/2791
* fix: incorrect timeline month display by @xwjdsh in https://github.com/usememos/memos/pull/2792
* feat: Enable iframe rendering in markdown code block by @MarkRushB in https://github.com/usememos/memos/pull/2799
* chore: update zh-Hans & zh-Hant translations by @moehanabi in https://github.com/usememos/memos/pull/2804
* chore: update zh-Hant by @ti777777 in https://github.com/usememos/memos/pull/2815
* fix: avoid making memos public when disabled by @moehanabi in https://github.com/usememos/memos/pull/2816
* fix: encode filename when using url prefix for resources by @mudkipme in https://github.com/usememos/memos/pull/2829
* chore: update i18n from Weblate by @boojack in https://github.com/usememos/memos/pull/2832
* fix: use mysql to parse dsn by @zwpaper in https://github.com/usememos/memos/pull/2838
* feat: add buf plugin to generate openapiv2 spec by @lincolnthalles in https://github.com/usememos/memos/pull/2843

## New Contributors
* @MarkRushB made their first contribution in https://github.com/usememos/memos/pull/2799
* @moehanabi made their first contribution in https://github.com/usememos/memos/pull/2804
* @mudkipme made their first contribution in https://github.com/usememos/memos/pull/2829
* @zwpaper made their first contribution in https://github.com/usememos/memos/pull/2838

**Full Changelog**: https://github.com/usememos/memos/compare/v0.18.2...v0.19.0

<hr /><em>This discussion was created from the release <a href='https://github.com/usememos/memos/releases/tag/v0.19.0'>v0.19.0</a>.</em>

**Comment by boojack** [comment]
## New syntax for embedding content

We've introduced a new feature that allows you to embed content. Now, you can easily embed content using the following syntax:

```markdown
![[memos/memoid]]
```

https://github.com/usememos/memos/assets/24653555/771b566d-a9ca-4002-9f5e-b4dfe914b116

## Inline mode of embedded content

Futher more, you can also seamlessly embed content directly with adding a `inline` parameter.

```markdown
![[memos/memoid?inline]]
```

https://github.com/usememos/memos/assets/24653555/d83348c0-2582-4f71-a3b0-9b2cc5f98382

**Comment by boojack** [comment]
## Rendering HTML with codeblock

With the addition of support for the `__html` language in codeblock, you can now embed HTML elements, YouTube videos, and more directly into your content. This enhancement offers greater versatility, allowing you to create richer and more dynamic presentations with ease.

```__html
<b>This is BOLD text</b>
```

https://github.com/usememos/memos/assets/24653555/732a75fe-2a49-4cd4-9691-e5f59a626b87

**Comment by boojack** [comment]
## Rename your TAG

Tag renaming has been brought up many times. Previously, we couldn't accurately determine which tags were associated with content, potentially leading to incorrect content updates. Now, with our new markdown parser, we can swiftly and accurately rename tags across all content.

https://github.com/usememos/memos/assets/24653555/fc7e4f14-c24e-4290-9f50-16466b38587b

**Comment by ccinoo** [comment]
How can the old version be upgraded to the latest v0.19.0 version? Is there any relevant tutorial available? Thanks

**Reply by boojack** [reply]
Learn more in docs: https://www.usememos.com/docs/install/self-hosting#upgrading-memos

**Reply by diegoaragunde** [reply]
This is not an upgrade but a backup from your previous DB. When you try to start 0.19 with this DB, Memos start consuming all the memory and hangs the host.
What is needed is a clear instruction how to migrate (not to backup) the DB from 0.18 to 0.19 in order to run Memos properly and in a healthy way.

**Comment by lincolnthalles** [comment]
> This is not an upgrade but a backup from your previous DB. When you try to start 0.19 with this DB, Memos start consuming all the memory and hangs the host. What is needed is a clear instruction how to migrate (not to backup) the DB from 0.18 to 0.19 in order to run Memos properly and in a healthy way.

The backup is just for extra safety.

Can you elaborate on that issue? Which is the exact previous version (0.18.0, 0.18.1 or 0.18.2), how big is your database, which type of database driver are you using, and how much is "all the memory"?

Memos usually handles upgrades nicely, and without user intervention. You just start a new version with an older database and it migrates the schema, if needed.
